runtime/runtime/src/lib.rs

```diff
@@ -1224,7 +1239,7 @@ impl Runtime {
                 outgoing_receipts.push(receipt);
             }
 
-            total_gas_burnt += outcome_with_id.outcome.gas_burnt;
+            total_gas_limit_used += outcome_with_id.outcome.gas_burnt;
```

Comment:
it seems a bit weird to add bunt gas to used gas here and elsewhere, especially considering those are 2 different fields of ActionResult

Reply:
The variable name is to keep track of total gas burnt for the purpose of gas limit, it has nothing to do with gas_used. I agree that this implementation will make the code a little confusing. That's because gas_burnt is used for two purposes, one to keep account the real gas burnt to charge the signer, one for keeping track of gas limit, and these two concepts are different for refund receipts, but they are stored in one field `gas_burnt`. Another solution is to add another field in ActionResult and ExecutionOutcome specifically for the use of gas limit tracking. I thought that might be overkill and that ExecutionOutcome is stored as part of Chain state which I am not super familiar with, so I don't know what effect that will bring there.

Reply:
I think maybe rename leaving variable name as it was (total_gas_burnt) or use some other name (like total_execution_gas) to disambiguate with result.gas_used